components/ai/code/ai-bbe/natural_programming_const_compile_time_code_generation_code.md

```diff
@@ -1,19 +1,25 @@
 ```
-import ballerina/http;
 import ballerina/test;
 
 function getTestProducts() returns Product[] =>
     const natural {
-        Generate list of products with:
-        - id: random id starts with "PROD", example:-  "PROD001"
+        Generate a list of products with:
+        - id: random ID that starts with "PROD". E.g., "PROD001"
         - price: random decimal between 10.0 and 1000.0
     };
 
 @test:Config
 function testFilterProducts() returns error? {
-    Product[] filteredProducts = filterProductsAbovePrice(getTestProducts(), 100);
-    foreach Product product in filteredProducts {
-        test:assertTrue(product.price > 100d);
+    Product[] testProducts = getTestProducts();
+    Product[] filteredProducts = filterProductsAbovePrice(testProducts, 100);
+    foreach Product product in testProducts {
+        int? index = filteredProducts.indexOf(product);
+        if product.price > 100d {
+            test:assertTrue(index != ());
+            continue;
+        }
+
+        test:assertTrue(index == ());
```

Comment:
Use assertEquals instead of assertTrue.

An if-else is okay here too, since the `continue` needs to be added without.